askametric/query_processor/query_processor.py

```diff
@@ -267,10 +292,15 @@ async def _run_data_analysis(self) -> None:
             self.status = ProcessorStatus.INTERNAL_ERROR
 
     @track_time(create_class_attr="timings")
-    async def process_query(self) -> None:
+    async def process_query(self, api_key: str | None = None) -> None:
         """
         The function processes the user query and returns the final answer.
+
+        Args:
+            api_key (str or None): (Optional) API key to use for the LLM call
         """
+        self._api_key = api_key
```

Comment:
We avoid saving the API key as a class attribute, while ensuring we pass the same API key throughout the pipeline by

1. Temporarily save api_key as a private class attribute (this line)
2. run all the private class functions (below this line)
3. Unset the private _api_key attr to None